src/commands.js

```diff
@@ -254,15 +255,17 @@ export const editorCommands = {
 
     // Change tags
     setTag(editor, tagName) {
+        editor.preventHistoryStep();
         const restoreCursor = preserveCursor(editor.document);
         const selectedBlocks = [...new Set(getTraversedNodes(editor.editable).map(closestBlock))];
-        for (const selectedBlock of selectedBlocks) {
-            const block = closestBlock(selectedBlock);
+        for (const block of selectedBlocks) {
             if (
                 ['P', 'PRE', 'H1', 'H2', 'H3', 'H4', 'H5', 'H6', 'BLOCKQUOTE'].includes(
                     block.nodeName,
                 )
             ) {
+                setCursor(block, 0, block, nodeSize(block));
+                editor.document.execCommand('removeFormat');
```

Comment:
1. I would write `editor.preventHistoryStep();` and `editor.restoreHistoryStep();` above and under this line as it's the only one concerned and make the understanding of the scope simple and explicit.

Reply:
Sure why not. The idea was to wrap the command between these two to show that we're going to do _one_ operation and prevent steps in the middle whatever happens.